EIPS/eip-abstract_wallet_transaction_api.md

```diff
@@ -0,0 +1,218 @@
+up)
+transaction.
+
+This is made significantly worse with account abstracted wallets (e.g. [EIP-4337](./eip-4337.md)
+or [EIP-3074](./eip-3074.md)), where the actual transaction hash may not be known at the time of signing but must be
+returned by `eth_sendTransaction`.
+
+Dapps need a better interface for sending messages from the user's wallet so the dapp can interact with wallets without
+worrying about the differences between wallet implementations. Some wallets may use [EIP-3074](./eip-3074.md) to deliver
+batches of messages as a single transaction, whereas others will use smart contract accounts
+via [EIP-4337](./eip-4337.md).
+The goal of the new API is to be a more stable and flexible interface while the wallet implementations evolve over time.
+
+## Specification
+
+Two new JSON RPC methods are added. Dapps may begin using these methods immediately, falling back
+to `eth_sendTransaction`
+and `eth_getTransactionReceipt` when they are not available.
+
+### `wallet_sendMessageBundle`
+
+Requests that the wallet deliver a group of messages on-chain from the user's wallet.
+The wallet MAY send all the messages as a single transaction.
+Dapps MUST NOT rely on the messages being sent in a single transaction, i.e. other untrusted transactions may be
+included between each of the transactions.
+The wallet MUST send these messages in the order specified in the request.
+The wallet MUST attempt to deliver all messages if it returns a successful response to this method, and the wallet MUST
+NOT deliver any
+messages if it returns an error response.
+The wallet MAY reject the request if the chain ID does not match the currently selected chain ID.
+The wallet MUST send the messages on the specified chain ID.
+The wallet MUST reject the request if the `from` address does not match the enabled account.
+The wallet MAY reject the request if one or more messages in the bundle will fail.
+
+#### Parameters
+
+The method takes an array containing one object element, with the following keys:
+
+- `chainId`: `INTEGER` - The chain ID on which the transactions should be sent
+- `from`: `ADDRESS` - The address of the wallet that should send the messages in the bundle
+- `messages`: `ARRAY` - An array containing 1 or more message objects, where each message has the following keys:
+    - `to`: `ADDRESS` - (optional when creating new contract) The address the message is directed to.
+    - `gas`: `QUANTITY` - (optional) Integer of the gas provided for the message execution. Unused gas MAY be made
+      available to subsequent messages.
+    - `value`: `QUANTITY` - (optional) Integer of the value sent with this transaction.
+    - `data`: `DATA` - The compiled code of a contract OR the message data to include with the call, e.g. encoded
+      function signature and parameters.
+    - `optional`: `BOOLEAN` - (optional, default false) Whether the wallet should continue with subsequent messages if
+      this particular message fails
```

Comment:
Can you specify this in the same way as the [`execution-apis`](https://github.com/ethereum/execution-apis/blob/main/src/eth/submit.yaml)?

Reply:
created a pr documenting these methods https://github.com/ethereum/execution-apis/pull/317